scripts/cat/thoughts.py

```diff
@@ -94,7 +95,36 @@ def cats_fulfill_thought_constraints(main_cat, random_cat, thought, game_mode, b
                 return False
         elif 'random_status_constraint' in thought and not random_cat:
             pass
+        
+        # Constraints for empty positions of power in the Clan
+        # Only works for leader, deputy, and medicine cat
+        if 'clan_missing_status_constraint' in thought:
```

Comment:
I'd like to see this be applicable to all ranks and I believe it would be relatively simple.  The PatrolEvent and ShortEvent formats already utilize a similar constraint system, however theirs is done via tags and then is filtered via the `get_alive_status_cats()` function in `utility.py`.  Plugging the list from `clan_missing_status_constraint` into `get_alive_status_cats()` would work well, since the function already takes a list of status as a kwarg.

It might additionally be wise to create a `clan_present_status_constraint` to handle the opposite situations: where you want to check that a certain status *is* present in the clan.

Reply:
Wow, thank you! I had no idea that utility function existed. Now it works much more smoothly!